core/src/test/java/kafka/server/share/SharePartitionManagerTest.java

```diff
@@ -0,0 +1,916 @@
+        shareFetchData.put(new TopicIdPartition(topicId, new TopicPartition("foo", 0)),
+                new ShareFetchRequest.SharePartitionData(topicId, 4000));
+        assertThrows(InvalidRequestException.class,
+                () -> sharePartitionManager.newContext("grp", shareFetchData, new ArrayList<>(), new ShareFetchMetadata(Uuid.ZERO_UUID, -1)));
+
+        // shareFetchData is not empty, but the maxBytes of topic partition is 0, which means this is added only for acknowledgements.
+        // New context should be created successfully
+        shareFetchData.clear();
+        shareFetchData.put(new TopicIdPartition(topicId, new TopicPartition("foo", 0)),
+                new ShareFetchRequest.SharePartitionData(topicId, 0));
+        shareFetchContext = sharePartitionManager.newContext("grp", shareFetchData, new ArrayList<>(), newReqMetadata);
+        assertEquals(shareFetchContext.getClass(), SharePartitionManager.FinalContext.class);
+    }
+
+    @Test
+    public void testNewContext() {
+        Time time = new MockTime();
+        ShareSessionCache cache = new ShareSessionCache(10, 1000);
+        SharePartitionManager sharePartitionManager = SharePartitionManagerBuilder.builder()
+                .withCache(cache).withTime(time).build();
+        Map<Uuid, String> topicNames = new HashMap<>();
+        Uuid tpId0 = Uuid.randomUuid();
+        Uuid tpId1 = Uuid.randomUuid();
+        topicNames.put(tpId0, "foo");
+        topicNames.put(tpId1, "bar");
+        TopicIdPartition tp0 = new TopicIdPartition(tpId0, new TopicPartition("foo", 0));
+        TopicIdPartition tp1 = new TopicIdPartition(tpId0, new TopicPartition("foo", 1));
+        TopicIdPartition tp2 = new TopicIdPartition(tpId1, new TopicPartition("bar", 0));
+        TopicIdPartition tp3 = new TopicIdPartition(tpId1, new TopicPartition("bar", 1));
+
+        String groupId = "grp";
+
+        // Verify that final epoch requests get a FinalContext
+        ShareFetchContext context1 = sharePartitionManager.newContext(groupId, new HashMap<>(), emptyPartList,
+                new ShareFetchMetadata(Uuid.randomUuid(), ShareFetchMetadata.FINAL_EPOCH));
+        assertEquals(context1.getClass(), SharePartitionManager.FinalContext.class);
+
+        // Create a new share session with an initial share fetch request
+        Map<TopicIdPartition, ShareFetchRequest.SharePartitionData> reqData2 = new LinkedHashMap<>();
+        reqData2.put(tp0, new ShareFetchRequest.SharePartitionData(tp0.topicId(), 100));
+        reqData2.put(tp1, new ShareFetchRequest.SharePartitionData(tp1.topicId(), 100));
+
+
+        ShareFetchMetadata reqMetadata2 = new ShareFetchMetadata(Uuid.randomUuid(), ShareFetchMetadata.INITIAL_EPOCH);
+        ShareFetchContext context2 = sharePartitionManager.newContext(groupId, reqData2, emptyPartList, reqMetadata2);
+        assertEquals(context2.getClass(), ShareSessionContext.class);
```

Comment:
signature is: `assertEquals(expected, actual)`

```suggestion
        assertEquals(ShareSessionContext.class, context2.getClass());
```